src/github.com/couchbase/sync_gateway/db/event.go

```diff
@@ -143,9 +144,13 @@ func (ef *JSEventFunction) CallFunction(event Event) (interface{}, error) {
 	switch event := event.(type) {
 
 	case *DocumentChangeEvent:
-		result, err = ef.Call(event.Doc)
+		if (len(event.OldDoc) == 0) {
+			result, err = ef.Call(event.Doc, nil)
+		} else {
+			result, err = ef.Call(event.Doc, event.OldDoc)
+		}
```

Comment:
In the sync function processing, it looks like we're explicitly turning oldDoc into an sgbucket.JSONString before including in Call.  Should we be doing that here as well?